pylint/extensions/typing.py

```diff
@@ -219,6 +224,66 @@ def visit_attribute(self, node: nodes.Attribute) -> None:
         if self._should_check_callable and node.attrname == "Callable":
             self._check_broken_callable(node)
 
+    @only_required_for_messages("redundant-union-assign-typehint")
+    def visit_annassign(self, node: nodes.AnnAssign) -> None:
+        type_annotations = list(node.annotation.get_children())
+        if not type_annotations:
+            return
+        if (
+            isinstance(type_annotations[0], nodes.Name)
+            and type_annotations[0].name in UNION_NAMES
+        ):
```

Comment:
What I mentioned earlier, it's difficult to follow what `type_annotations[0]` is referring to. `node.annotation.value` would be better.